docker-compose-files/dspace-compose/config-definition.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0"?>
+<!--
+     DSpace Configuration Definition
+     This Apache Commons Configuration definition file defines the ordering of configuration files loaded by DSpace.
+     Optionally, you may choose to add additional configuration files/locations.
+
+     For more information and examples of valid configuration definition files, see:
+     https://commons.apache.org/proper/commons-configuration/userguide/howto_combinedbuilder.html
+-->
+<configuration>
+    <!-- Configurations in this section override one another.
+         Earlier values take precedence, and override any values in later config files -->
+    <override>
+        <!-- By default, we load all System Properties and Environment variables, in case any override default settings
+             (e.g. "-Ddspace.dir" will override "dspace.dir"). Optionally, you could choose to disable this setting, but
+             still load individual settings from System or Env using variable interpolation,
+             e.g. ${sys:dspace.dir} or ${env:DSPACE_HOME}
+             For more info see:
+             https://commons.apache.org/proper/commons-configuration/userguide/howto_basicfeatures.html#Variable_Interpolation -->
+        <!-- Load Java System properties (does not automatically reload) -->
+        <system/>
+        <!-- Load Environment variables (does not automatically reload) -->
+        <env/>
+
+        <!-- Allows docker insertion of properties containing periods from environment variables -->
+        <properties fileName="environment.cfg" throwExceptionOnMissing="false" config-optional="true"
```

Comment:
@terrywbrady : I have to admit, I'm a little hesitant to add an entirely new configuration file (`environment.cfg`) *just for* the Docker `ingestAIP.sh` script (which seems to be the only place this is used).  

I worry it's a bit complex to have two overriding config files (`local.cfg` and now `environment.cfg`) as DSpace users would need to clearly understand precedence of each. They'd also need to realize that an "environment.cfg" file in Docker will *override* any changes they try to make in the "local.cfg" -- which could get very frustrating, if they are used to changing "local.cfg" and find Docker isn't respecting those changes any longer.

Also, it seems like `ingestAIP.sh` script only gets run *once*, which would imply that changes to environment variables are not being reloaded into your `environment.cfg`.  So, after initial startup of Docker, you would need to modify the `environment.cfg` file directly, and ignore anything set in environment variables.

I'm sorry to say this, but honestly, if we only really need to customize ~15 settings (`dspace.name`, `db.*` settings, and `rdf.*` settings) during Docker startup, then I still feel we may be better off using `JAVA_OPTS`.  I'm worried about this approach for the reasons mentioned above.